### PR TITLE
feat: add Node 18 support

### LINKED ISSUE #2205: Add Node 18 support
### New Feature / Enhancement Checklist
- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-dashboard/issues?q=is%3Aissue).

### Current Limitation
n/a

### Feature / Enhancement Description
Include the current (soon active) version of NodeJS version 18

### Example Use Case
n/a

### Alternatives / Workarounds
n/a

### 3rd Party References
n/a

### PR DESCRIPTION
**Pull request by MrMartinR**
### New Pull Request Checklist
- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-dashboard/issues?q=is%3Aissue).

### Issue Description
Add Node18 from CI test

Related issue: #[2205](https://github.com/parse-community/parse-dashboard/issues/2205)

### Approach
n/a

### TODOs before merging
- [x] Add tests
- [ ] Add changes to documentation (guides, repository pages, in-code descriptions)
- [x] A changelog entry is created automatically using the pull request title (do not manually add a changelog entry)

### CODE REVIEW

**Comment by MrMartinR** [comment]
Hi @parse-community/dashboard, can you review this PR?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ on:
     branches:
       - '**'
 env:
-  NODE_VERSION: 16.10.0
+  NODE_VERSION: 18.0.0
```
Comment (mtrezza) [inline .github/workflows/ci.yml]: Use latest node version, https://github.com/nodejs/node/blob/main/doc/changelogs/CHANGELOG_V18.md

**Review by mtrezza — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (MrMartinR) [reply .github/workflows/ci.yml]: wth.. I was looking for it but I couldn't find it...!

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -143,12 +145,6 @@ jobs:
           key: ${{ runner.os }}-node-${{ matrix.NODE_VERSION }}-${{ hashFiles('**/package-lock.json') }}
           restore-keys: |
               ${{ runner.os }}-node-${{ matrix.NODE_VERSION }}-
-      - name: Install dependencies (Node < 10)
```
Comment (mtrezza) [inline .github/workflows/ci.yml]: Unrelated change, this should be done when removed node 12

**Review by mtrezza — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (MrMartinR) [reply .github/workflows/ci.yml]: I was not sure about this, but all the versions are > 10 isn't?

Comment (MrMartinR) [reply .github/workflows/ci.yml]: I am gonna put it back

Comment (mtrezza) [reply .github/workflows/ci.yml]: Yes, it's more suitable to remove it when removing the unsupported Node versions.

**Review by mtrezza — CHANGES_REQUESTED** [review]
Can you also set the default Node version to 18?

env:
  NODE_VERSION: 16.10.0

**Comment by MrMartinR** [comment]
the default version to 18 or 16?

**Comment by mtrezza** [comment]
from 16 to 18 latest

**Review by mtrezza — CHANGES_REQUESTED** [review]
change default version from 16 to 18 latest

**Comment by MrMartinR** [comment]
Changed "again?" default version to Node 18.4.0

**Comment by mtrezza** [comment]
Not sure what you mean by "again?"... but the changes look correct now. Let's see if the CI passes.

**Comment by mtrezza** [comment]
Node 18 fails with:

```
/bin/tar xz --warning=no-unknown-keyword -C /home/runner/work/_temp/f0db3245-e55c-4cee-8109-46364e0be05d -f /home/runner/work/_temp/cf3cd178-2a64-47da-9f4b-43abbd53dae6
/opt/hostedtoolcache/node/18.4.0/x64/bin/node --version
/opt/hostedtoolcache/node/18.4.0/x64/bin/node: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by /opt/hostedtoolcache/node/18.4.0/x64/bin/node)
Error: The process '/opt/hostedtoolcache/node/18.4.0/x64/bin/node' failed with exit code 1
```

This is due to [toolchain-and-compiler-upgrades](https://github.com/nodejs/node/blob/main/doc/changelogs/CHANGELOG_V18.md#toolchain-and-compiler-upgrades).

Could you change every `ubuntu-18.04` to `ubuntu-latest`?

**Comment by MrMartinR** [comment]
> Not sure what you mean by "again?"... but the changes look correct now. Let's see if the CI passes.

I was confused because I changed that already in some commit before you asked, so I was not sure If you mean to keep the 16 or 18.. 🤔

**Comment by MrMartinR** [comment]
Looks like the test passed with the `ubuntu-latest`

**Comment by mtrezza** [comment]
Yes pretty neat! Couod you also make the necessary changes in the Readme? There is a compatibility table.

**Comment by MrMartinR** [comment]
Bumped to 18.5.0 and added the node 18 to the README.

**Review by mtrezza — APPROVED** [review]
Looks good! Well done; did you actually try out the dashboard in a Node 18 environment? The challenge with Parse Dashboard is that it doesn't have unit tests, unlike Parse Server; so everything requires some manual testing before merging.

**Comment by MrMartinR** [comment]
> Looks good! Well done; did you actually try out the dashboard in a Node 18 environment? The challenge with Parse Dashboard is that it doesn't have unit tests, unlike Parse Server; so everything requires some manual testing before merging.

Hi, no, actually I haven't, I once was using nvm (node version manager) but I ended messing up my project, months ago I upgraded to node 18 but I had to come back to node 16, not sure when I will try again.
Also no idea about programming unit test 😅

**Review by mtrezza — CHANGES_REQUESTED** [review]
Requires manually testing dashboard in Node 18, as there are no unit tests that can verify that dashboard runs in Node 18 env.

**Comment by MrMartinR** [comment]
I upgraded to Node v18 in my local environment.
```
Node Sass does not yet support your current environment: OS X 64-bit with Unsupported runtime (108)
For more information on which environments are supported please see:
https://github.com/sass/node-sass/releases/tag/v7.0.0
 ```
Lots of errors trying to run parse-dashboard on Node v18
Looks like node-sass is deprecated and they recommend to move to Dart-sass? Not sure if they will include Node v18 support for Node Sass or not...

**Comment by mtrezza** [comment]
You'd have to do some research to figure out more. Others may face the same issue and there may be solutions out there that could guide you. It's also possible that this will be fixed until Node 18 enters active LTS.

**Comment by MrMartinR** [comment]
Following this PR https://github.com/sass/node-sass/pull/3257 I assume in the next release Node 18 will be supported, the other option will be replace node-sass for dart-sass

**Comment by mtrezza** [comment]
I tried it out in a Node 18 env and it worked without any issues.

**Review by mtrezza — APPROVED** [review]
Looks good!

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [4.2.0-alpha.10](https://github.com/parse-community/parse-dashboard/releases/tag/4.2.0-alpha.10)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [4.2.0-beta.2](https://github.com/parse-community/parse-dashboard/releases/tag/4.2.0-beta.2)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [4.2.0](https://github.com/parse-community/parse-dashboard/releases/tag/4.2.0)
